src/NuGet.Core/NuGet.Build.Tasks/BuildTasksUtility.cs

```diff
@@ -227,5 +228,90 @@ public static void AddPropertyIfExists(IDictionary<string, string> properties, s
+        /// If the value of <paramref name="restoreProjectStyle"/> is not empty and could not be parsed, <code>null</code> is returned.</returns>
+        internal static (ProjectStyle ProjectStyle, bool IsPackageReferenceCompatibleProjectStyle) GetProjectRestoreStyle(string restoreProjectStyle, bool hasPackageReferenceItems, string projectJsonPath, string projectDirectory, string projectName, Common.ILogger log)
+        {
+            ProjectStyle projectStyle;
+
+            // Allow a user to override by setting RestoreProjectStyle in the project.
+            if (!string.IsNullOrWhiteSpace(restoreProjectStyle))
+            {
+                if (!Enum.TryParse(restoreProjectStyle, ignoreCase: true, out projectStyle))
+                {
+                    // Any value that is not recognized is treated as Unknown
+                    projectStyle = ProjectStyle.Unknown;
+                }
+            }
+            else if (hasPackageReferenceItems)
+            {
+                // If any PackageReferences exist treat it as PackageReference. This has priority over project.json.
+                projectStyle = ProjectStyle.PackageReference;
+            }
+            else if (!string.IsNullOrWhiteSpace(projectJsonPath))
+            {
+                // If this is not a PackageReference project check if project.json or projectName.project.json exists.
+                projectStyle = ProjectStyle.ProjectJson;
+            }
+            else if (ProjectHasPackagesConfigFile(projectDirectory, projectName))
+            {
+                // If this is not a PackageReference or ProjectJson project check if packages.config or packages.ProjectName.config exists
+                projectStyle = ProjectStyle.PackagesConfig;
+            }
+            else
+            {
+                // This project is either a packages.config project or one that does not use NuGet at all.
+                projectStyle = ProjectStyle.Unknown;
+            }
+
+            bool isPackageReferenceCompatibleProjectStyle = projectStyle == ProjectStyle.PackageReference || projectStyle == ProjectStyle.DotnetToolReference;
+
+            return (projectStyle, isPackageReferenceCompatibleProjectStyle);
+        }
+
+        /// <summary>
+        /// Determines if the project has a packages.config file.
+        /// </summary>
+        /// <param name="projectDirectory">The full path of the project directory.</param>
+        /// <param name="projectName">The name of the project file.</param>
+        /// <returns><code>true</code> if a packages.config exists next to the project, otherwise <code>false</code>.</returns>
+        private static bool ProjectHasPackagesConfigFile(string projectDirectory, string projectName)
+        {
+            if(string.IsNullOrWhiteSpace(projectDirectory))
+            {
+                throw new ArgumentNullException(nameof(projectDirectory));
```

Comment:
`ArgumentException`.  `ArgumentNullException` is only for when the argument is `null`, not an empty string or whitespace only.